components/Slider.js

```diff
@@ -0,0 +1,246 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+class Slider extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    id: PropTypes.string,
+    onChange: PropTypes.func,
+    value: PropTypes.number.isRequired,
+    min: PropTypes.number.isRequired,
+    minLabel: PropTypes.string,
+    max: PropTypes.number.isRequired,
+    maxLabel: PropTypes.string,
+    step: PropTypes.number,
+    stepMuliplier: PropTypes.number,
+    children: PropTypes.node,
+    disabled: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    stepMuliplier: 4,
+    disabled: false,
+    minLabel: '',
+    maxLabel: '',
+  }
+
+  state = {
+    dragging: false,
+    value: this.props.value,
+    left: 0,
+  };
+
+  componentDidMount() {
+    this.updatePosition()
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps !== this.props) {
+      this.updatePosition()
+    }
+  }
+
+  updatePosition = (evt) => {
+    if (evt && this.props.disabled) {
+      return;
+    }
+
+    if(evt) {
+      if(evt.dispatchConfig) { evt.persist(); } // fix
+    }
+
+    if (this.state.dragging) {
+      return;
+    }
+
+    this.setState({ dragging: true });
+
+    requestAnimationFrame(() => {
+      this.setState({ dragging: false });
+      const {
+        left,
+        newValue,
+      } = this.calcValue(evt);
+
+      this.setState({
+        left,
+        value: newValue,
+       });
+    });
+  }
+
+  calcValue = (evt) => {
```

Comment:
this is kind of a general comment on this function (and I know this probably should've been done on the vanilla version) - but this function is... pretty long. There are a couple sections where it might help readability if it was broken into a couple different functions. 

At the end of the day, you're calculating the `left` and `newValue` values here - you could think about breaking the `mousemove | click | touchmove` into a function, same for `keydown`